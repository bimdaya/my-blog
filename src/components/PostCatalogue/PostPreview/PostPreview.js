import React from 'react';
import Link from 'next/link';

import postPreviewStyles from './PostPreview.module.css';

const PostPreview = ({ slug, title, date, description }) => {
  return (
    <div className={postPreviewStyles.post_margin}>
      <h4>
        <Link as={`/posts/${slug}`} href="/posts/[slug]">
          <a className={postPreviewStyles.post_name}>
            ðŸ“‹ {title},<span className="post_published_date"> ðŸ“† {date}</span>
          </a>
        </Link>
      </h4>
      <p className={postPreviewStyles.post_description}>{description}</p>
    </div>
  );
};

export default PostPreview;
