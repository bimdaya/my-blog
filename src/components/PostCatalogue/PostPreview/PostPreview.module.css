.post_name {
  text-decoration: none !important;
  color: beige;
  font-size: 1vw;
}

.post_description {
  font-size: 1vw;
}

.post_margin {
  margin-bottom: 45px;
  margin-top: 90px;
}
