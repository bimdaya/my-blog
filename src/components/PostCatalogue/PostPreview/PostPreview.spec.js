import React from 'react';
import { render } from '@testing-library/react';
import PostPreview from './PostPreview';

describe('PostPreview', () => {
  it('renders', () => {
    const post = {
      slug: 'test_slug',
      title: 'test title',
      date: '2020.04.05',
      description: 'test description',
    };
    render(<PostPreview {...post} />);
  });
});
