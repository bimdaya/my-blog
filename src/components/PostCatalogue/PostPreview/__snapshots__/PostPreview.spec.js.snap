// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPreview renders 1`] = `
<div
  class="post_margin"
>
  <h4>
    <a
      class="post_name"
      href="/posts/test_slug"
    >
      ðŸ“‹ 
      test title
      ,
      <span
        class="post_published_date"
      >
         ðŸ“† 
        2020.04.05
      </span>
    </a>
  </h4>
  <p
    class="post_description"
  >
    test description
  </p>
</div>
`;
